README.md

```diff
@@ -315,3 +316,63 @@ Comp 40 gives you the tools to explore a large number of other topics:
+* Make sure you go to class. Mark does not use a powerpoint, so if you miss one, then you will not have any resources to catch up with class work. So if something happens, make sure you get note from someone and try to never fall behind because the class goes by quick!
+* There is no more HalliganHelper, but Piazza is always on to communicate with the TAs. Identify the location of the TAs through Piazza to get help! Also, Mark holds regular office hours and he is of great help if you want a deep understanding of the concepts discussed in class. From my personal experience, he normally takes around 20~30 minutes to explain concepts each time I visited, so if you are in line, you may have to wait quite a time. But once it is your turn, you will get the thorough help that you may need. Mark is awesome during office hours, so do try to take advantage of it!
+* Comp 15 debugging tutorial 101:
+  * First, you should run your code through a debugger (gdb) and/or valgrind. If you don't know how to use one or both of these tools, it is absolutely worth your time to meet with a TA at the beginning of the semester to get an introduction to these tools. Memory management is new to all comp 11 students and it is essential you learn how to use these tools. You won’t survive without it, so learn it ASAP!
+  * Humans are bad at reasoning about code, so you need to have some idea of what is actually going on in your code. Hence, when you see errors, you should draw pictures and bring them to the TAs to clarify both what you intended and what is actually going on in your code. This will help you make a better educated guess and give the TAs a lot more pleasant experience.
+  * You should explain to the TA what you have tried already. "My code is broken" and "Why is this segfault happening?" are common statements that a TA hears when they first walk up to a group, but are not helpful places to begin.
+* Exams and projects are difficult, but Mark is very understanding. Don’t freak out and know it’s okay to have a low score on the exam because Mark will curve well. He understands that he is throwing a lot at you and will reward you for all your hard work. You may very well be surprised with your final score. For projects in particular, he will likely extend the deadline if the difficulty is too high, so keep working hard and please stay healthy :)
+
+### Combining Material
```

Comment:
This section does not get much into combining & extending material. See COMP 105 -- most of the stuff in there is not covered, but is an extension of course material. The following section should be completely unrelated stuff that is not covered.

A good example for this section would be the 20-odd different types of trees commonly used, etc.